packages/core/tests/integration/protocol-versions.test.ts

```diff
@@ -0,0 +1,362 @@
+/** biome-ignore-all lint/style/noNonNullAssertion: tests */
+import { describe, expect, test } from "bun:test";
+import { z } from "zod";
+import { buildInitializeRequest, createStatefulTestServer } from "../utils.js";
+
+describe("Protocol Version Negotiation", () => {
+  test("should accept 2025-03-26 and echo it back", async () => {
+    const { handler } = createStatefulTestServer();
+
+    const initRequest = new Request("http://localhost:3000/", {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({
+        jsonrpc: "2.0",
+        id: 1,
+        method: "initialize",
+        params: {
+          clientInfo: { name: "test-client", version: "1.0.0" },
+          protocolVersion: "2025-03-26",
+          capabilities: {},
+        },
+      }),
+    });
+
+    const response = await handler(initRequest);
+    expect(response.status).toBe(200);
+
+    const result = await response.json();
+    expect(result.result.protocolVersion).toBe("2025-03-26");
+  });
+
+  test("should accept 2025-06-18 and echo it back", async () => {
+    const { handler } = createStatefulTestServer();
+
+    const initRequest = buildInitializeRequest();
+    const response = await handler(initRequest);
+    expect(response.status).toBe(200);
+
+    const result = await response.json();
+    expect(result.result.protocolVersion).toBe("2025-06-18");
+  });
+
+  test("should reject unsupported protocol version", async () => {
```

Comment:
question about this test: is this really according to spec? 

my understanding from reading is that the server simply echoes back the latest version it supports and then the client is responsible for disconnecting 

> The version of the Model Context Protocol that the server wants to use. This may not match the version that the client requested. **If the client cannot support this version, it MUST disconnect.**

i don't think that the initialize request should return an error per se?

https://modelcontextprotocol.io/specification/2025-06-18/schema#initializeresult

Reply:
fixed in latest